internal/compiler/generator/rust.rs

```diff
@@ -1064,6 +1064,41 @@ fn generate_sub_component(
         quote!(usize::MAX)
     };
 
+    let timer_names =
```

Comment:
To reduce the amount of generated code, you could consider placing the common code into a helper function like this: https://github.com/slint-ui/slint/blob/8a0df2b3a5b6e74e4c52f8177d20d33c7ab5aa54/api/rs/slint/private_unstable_api.rs#L53

Maybe it all can go into an array?

Reply:
I thought of it but didn't do it yet because it's a bit tricky with the `if` and stuff. But I think it can be improved later on.